filebeat/prospector/prospector_log.go

```diff
@@ -42,12 +43,33 @@ func NewLog(p *Prospector) (*Log, error) {
 	return prospectorer, nil
 }
 
+func (l *Log) resolvePaths() error {
+	if l.paths != nil {
+		return nil
+	}
+	if !l.config.recursiveGlob {
+		l.paths = l.config.Paths
+		return nil
+	}
+	l.paths = []string{}
```

Comment:
This is unnecessary if you check the `len()` of the slice above, rather than for non-nil.